token/program-2022-test/tests/cpi_guard.rs

```diff
@@ -0,0 +1,769 @@
+#![cfg(feature = "test-sbf")]
+
+mod program_test;
+use {
+    program_test::{keypair_clone, TestContext, TokenContext},
+    solana_program_test::{
+        processor,
+        tokio::{self, sync::Mutex},
+        ProgramTest,
+    },
+    solana_sdk::{
+        instruction::InstructionError, pubkey::Pubkey, signature::Signer, signer::keypair::Keypair,
+        transaction::TransactionError, transport::TransportError,
+    },
+    spl_instruction_padding::instruction::wrap_instruction,
+    spl_token_2022::{
+        error::TokenError,
+        extension::{
+            cpi_guard::{self, CpiGuard},
+            ExtensionType,
+        },
+        instruction::{self, AuthorityType},
+        processor::Processor as SplToken2022Processor,
+    },
+    spl_token_client::{
+        client::ProgramBanksClientProcessTransaction,
+        token::{Token, TokenError as TokenClientError},
+    },
+    std::sync::Arc,
+};
+
+// set up a bank and bank client with spl token 2022 and the instruction padder
+// also creates a token with no extensions and inits two token accounts
+async fn make_context() -> (TestContext, Pubkey) {
+    // TODO this may be removed when we upgrade to a solana version with a fixed `get_stack_height()` stub
+    if std::env::var("BPF_OUT_DIR").is_err() && std::env::var("SBF_OUT_DIR").is_err() {
+        panic!("CpiGuard tests MUST be invoked with `cargo test-sbf`, NOT `cargo test --feature test-sbf`. \
+                In a non-BPF context, `get_stack_height()` always returns 0, and all tests WILL fail.");
+    }
+
+    let instruction_padding_id = Keypair::new().pubkey();
+
+    let mut program_test = ProgramTest::new(
+        "spl_token_2022",
+        spl_token_2022::id(),
+        processor!(SplToken2022Processor::process),
+    );
+
+    program_test.add_program(
+        "spl_instruction_padding",
+        instruction_padding_id,
```

Comment:
up to you, but if you want to make this simpler and not require passing back the pubkey id as the result of this function, you can just use `spl_instruction_padding::id()` everywhere. Also, I've deployed it to all networks